internal/controller/streams.go

```diff
@@ -271,8 +280,39 @@ func (sm *streamManager) StreamFalcoNetworkFlows(ctx context.Context) error {
 	}
 }
 
-// connectAndStreamCiliumNetworkFlows creates networkFlowsStream client and begins the streaming of network flows.
-func connectAndStreamCiliumNetworkFlows(logger *zap.Logger, sm *streamManager) error {
+// StreamKeepalives loops infinitely as long as the keepalives are working. We
+// halt when the 'keepaliveDone' channel is written to, or when sending a
+// keepalive ping fails.
+//
+// This should be run inside a goroutine in every `connectAndStream*` function
+func (sm *streamManager) StreamKeepalives(
+	freq time.Duration,
+	sendKeepalive func(*streamManager) error,
+	keepaliveDone chan struct{},
+) {
+	defer func() {
+		keepaliveDone <- struct{}{}
+	}()
+
+	for {
+		timer := time.NewTimer(freq)
+		select {
+		case <-timer.C:
```

Comment:
1. Add jitter to the period.
1. Instead of creating a Timer on every iteration, you could reuse the same Timer and just Reset it.
1. Stop the Timer explicitly when returning, to reduce the number of goroutines.
```suggestion
	timer := time.NewTimer(jitterTime(period, 0.1)) // Add 10% jitter
	defer timer.Stop()

	for {
		select {
		case <-timer.C:
			timer.Reset(jitterTime(period, 0.1)) // Add 10% jitter
```

Reply:
A timer makes a whole new goroutine... of course. I knew this cuz I read the docs but I did not internalize it. Man. That feels like a silent efficiency killer. If I didn't `defer stop` the timer I'd leak a goroutine :/